src/status_im/contexts/wallet/add_account/create_account/import_private_key/private_key/view.cljs

```diff
@@ -86,11 +90,13 @@
 
 (defn view
   []
-  (let [theme                         (theme/use-theme)
-        customization-color           (rf/sub [:profile/customization-color])
-        [input-value set-input-value] (rn/use-state "")
-        [flow-state set-flow-state]   (rn/use-state)
-        error?                        (= :invalid-private-key flow-state)]
+  (let [theme                       (theme/use-theme)
+        customization-color         (rf/sub [:profile/customization-color])
+        private-key                 (rf/sub [:wallet.import-private-key/private-key])
+        public-address              (rf/sub [:wallet.import-private-key/public-address])
+        [flow-state set-flow-state] (rn/use-state)
+        error?                      (= :invalid-private-key flow-state)]
+    (rn/use-mount #(rf/dispatch [:wallet.import-private-key/clear-data]))
```

Comment:
this can be done on the navigation to this page instead? 🤔

Reply:
is there a particular reason for this @ibrkhalil?
@flexsurfer @ulisesmac is there a general preference to prefer to add db clean up events to when a user navigates away/ vs on mount?
The issue with on-mount is that it sometimes doesn't fire again if a user comes back to this page as the page has technically already mounted afaik, so it is easy to not have this correctly setup

Reply:
I don't know that this happened with `rn/use-mount`, From my experience I always used `(rn/use-unmount fn)` to clear previous form data and it worked really good.